flutter/ios/Classes/SwiftSentryFlutterPlugin.swift

```diff
@@ -3,248 +3,279 @@ import Sentry
-    }
-
-    if let dist = arguments["dist"] as? String {
-      options.dist = dist
-    }
-
-    if let enableAutoNativeBreadcrumbs = arguments["enableAutoNativeBreadcrumbs"] as? Bool,
-       enableAutoNativeBreadcrumbs == false {
-      options.integrations = options.integrations?.filter { (name) -> Bool in
-        name != "SentryAutoBreadcrumbTrackingIntegration"
-      }
-    }
-
-    if let maxBreadcrumbs = arguments["maxBreadcrumbs"] as? UInt {
-      options.maxBreadcrumbs = maxBreadcrumbs
-    }
-  }
-
-  private func logLevelFrom(diagnosticLevel: String) -> SentryLogLevel {
-    switch diagnosticLevel {
-    case "fatal", "error":
-      return .error
-    case "debug":
-      return .debug
-    case "warning", "info":
-      return .verbose
-    default:
-      return .none
+    
+    private func initNativeSdk(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
+        guard let arguments = call.arguments as? [String: Any], !arguments.isEmpty else {
+            print("Arguments is null or empty")
+            result(FlutterError(code: "4", message: "Arguments is null or empty", details: nil))
+            return
+        }
+        
+        SentrySDK.start { options in
+            self.updateOptions(arguments: arguments, options: options)
+            
+            self.sentryOptions = options
+            
+            // note : for now, in sentry-cocoa, beforeSend is not called before captureEnvelope
+            options.beforeSend = { event in
+                self.setEventOriginTag(event: event)
+                
+                if var sdk = event.sdk, self.isValidSdk(sdk: sdk) {
+                    if let packages = arguments["packages"] as? [String] {
+                        if var sdkPackages = sdk["packages"] as? [String] {
+                            sdk["packages"] = sdkPackages.append(contentsOf: packages)
+                        } else {
+                            sdk["packages"] = packages
+                        }
+                    }
+                    
+                    if let integrations = arguments["integrations"] as? [String] {
+                        if var sdkIntegrations = sdk["integrations"] as? [String] {
+                            sdk["integrations"] = sdkIntegrations.append(contentsOf: integrations)
+                        } else {
+                            sdk["integrations"] = integrations
+                        }
+                    }
+                    event.sdk = sdk
+                }
+                
+                return event
+            }
+        }
+        
+        if (startSession) {
+            // only this is not enough, we need to mimics SentrySessionTracker.didBecomeActive
+            // how to call that from here?
+            // it'd also be nice to know if theres already a running session, so we dont do twice, lets say that
+            // not always the SDK is init. too late.
+            SentrySDK.currentHub().startSession()
+        }
```

Comment:
after SDK is fully operational, in case that the Cocoa SDK missed that notification, auto session tracking is enabled, we initialize manually, cus we were able to listen to the notification before the Cocoa SDK.